oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -187,4 +211,44 @@ public APIGatewayProxyResponseEvent handleRequest(
                                                     authenticationResponse.toURI().toString()));
                         });
     }
+
+    private void updateClientSubjects(
+            UserProfile userProfile, AuthenticationRequest authenticationRequest) {
+        String clientId = authenticationRequest.getClientID().getValue();
+        ClientRegistry clientRegistry = dynamoClientService.getClient(clientId).get();
+
+        String sector =
+                clientRegistry.getSectorIdentifierUri() != null
+                        ? clientRegistry.getSectorIdentifierUri()
+                        : null;
+
+        if (clientRegistry.getSubjectType().equalsIgnoreCase("pairwise")) {
+            if (userProfile.getClientSubjects() != null) {
+                if (!clientSubjectsContainsClientId(userProfile.getClientSubjects(), clientId)) {
+                    userProfile
+                            .getClientSubjects()
+                            .add(new ClientSubject(clientId, new Subject(), sector));
+                } else {
+                    userProfile.setClientSubjects(
+                            List.of(new ClientSubject(clientId, new Subject(), sector)));
+                }
+                dynamoService.updateClientSubjectIds(
+                        userProfile.getEmail(), userProfile.getClientSubjects());
+            }
+        }
+    }
+
+    private boolean clientSubjectsContainsClientId(
+            List<ClientSubject> clientSubjects, String clientId) {
+        boolean present = false;
+
+        for (ClientSubject clientSubject : clientSubjects) {
```

Comment:
Should we be looking up on sector if client ID doesn't exist.

Reply:
I don't think we can have a ClientSubject entry without a client ID will have to confirm this with @sanisc